app/status.go

```diff
@@ -137,7 +137,9 @@ func GetUserStatusesByIds(userIds []string) ([]*model.Status, *model.AppError) {
 		}
 	}
 	for _, userId := range missingUserIds {
-		statusMap = append(statusMap, &model.Status{UserId: userId, Status: "offline"})
+		if result := <-Srv.Store.User().Get(userId); result.Err == nil {
+			statusMap = append(statusMap, &model.Status{UserId: userId, Status: "offline"})
```

Comment:
Yeah I think we just ignore the ones not in the DB or in the cache. A client has enough information to realize which ids they provided don't exist and can choose how to deal with that itself